unit2-apis-and-advanced-javascript/README.md

```diff
@@ -18,68 +18,68 @@
-  - **Learning outcome**: Given a class called Square and two instances of this class, s and t, how would you define `calcArea()` and `calcPermiter()` methods such that _they can be called_ on the two instantiated instances, s and t?
-  - **Learning outomce**: How would you reimplement jQuery from scratch using objects? (ie: `$('..')` gets us an object with methods such as `.css()`. How would you define this c;ass such that it doesn't require a `new` keyword *but still* consoles 'jQuery' when `console.log( $('...') instanceof jQuery )`
-  - **Learning outcome**: How would you extend you jQuery class such that it supports the creation of "plugins" (ie: a new method called `.fooPlugin()` that works seamlessly on all current and future instances of your jQuery obejct) _without touching your_ original jQuery class implementation source code
- 
-- APIs via AJAX and JSON: 2 days
+  - OOP, Prototypical OOP VS traditional class-based OOP, non-JS OOP (Interfaces, Abstract Classes)
+  - **Learning outcome**: Be able to discuss OOP interview questions (how to design games like chess, model animal species, etc.) and understand how to discuss the talking points for non-JS OOP
+  - **Learning outcome**: Design and implement classes in JS, create instances manually and dynamically, and utilize protypical inheritance to augment existing classes on the fly. [Examples here](https://github.com/C4Q/ac-curriculum-web/tree/master/unit2-apis-and-advanced-javascript/oo-challenges.md)
+- APIs via AJAX and JSON: 2 days (carries into Week 3)
   - What is AJAX?
   - jQuery's AJAX methods (ajax, get, post) & under-the-hood JavaScript's pure AJAX methods (only for educational reasons)
   - Handling success and failures with callbacks
   - Firebase (or some other competitor) configuration and consumption
-  - Promises: definition, usage, etc
   - How to perform multiple, ordered asynch calls using promises (Firebase DB reads and/or writes can be used here as example)
   - **Learning outcome**: Update the tic-tac-toe implementation from Unit 1 to store game state in Firebase, maintain leaderboards, have user accounts that users can log in to (insecurely) from any client
   - **Learning outcome**: Consume an existing API (http://jsonplaceholder.typicode.com/ or http://pokeapi.co/) and allow for the client to control different queries on the API which display different results 
 
+
 ### Week 3
 
 - Introduction to ReactJS: 2 days
   - React lifecycle methods: `render()`, `componentDidMount()`, `componentWillRecieveProps()`, `componentDidUpdate()`, etc
   - JSX and the benefits of writing HTML markup in javascript
   - Adding a reactjs task to your `gulpfile`
-  - Light introduction to ES6 (most of ReactJS + ReactJS documentation is written in es6)
-    - classes, extend keyword, arrow functions, spread operator
+  - Light introduction to ES6 (most of ReactJS + ReactJS documentation is written in es6) [_BENNO_: My concern is that there won't be enough time]
```

Comment:
No matter what we will put together resources for them to look at after class. But even their after-class time is precious, so we want to ration it out appropriately. We don't want to take away time from the content that we will be testing them on, that they will be tested on in interviews, and what will be crucial to building successful projects.

Let's leave it in, at the very least we can give them exposure.